packages/rum-core/test/common/compress.spec.js

```diff
@@ -0,0 +1,272 @@
+  span_count: 'yc',
+  stacktrace: 'st',
+  start: 's',
+  started: 'sd',
+  status_code: 'sc',
+  subtype: 'su',
+  sync: 'sy',
+  tags: 'g',
+  timeToFirstByte: 'fb',
+  trace_id: 'tid',
+  transaction: 'x',
+  transaction_id: 'xid',
+  'transaction.breakdown.count': 'xbc',
+  'transaction.duration.count': 'xdc',
+  'transaction.duration.sum.us': 'xds',
+  transfer_size: 'ts',
+  type: 't',
+  url: 'url',
+  user: 'u',
+  username: 'un',
+  value: 'v',
+  version: 've'
+}
+
+const SEPARATOR = '-'
+
+function flatten(obj) {
+  const result = {}
+  for (const key of Object.keys(obj)) {
+    const value = obj[key]
+    if (value != null && typeof value === 'object') {
+      const flatObj = flatten(value)
+      for (let flatKey of Object.keys(flatObj)) {
+        result[key + SEPARATOR + flatKey] = flatObj[flatKey]
+      }
+    } else {
+      result[key] = value
+    }
+  }
+  return result
+}
+
+describe('Compress', function() {
+  let serviceFactory
+  let apmServer
+  let configService
+  let performanceMonitoring
+  let errorLogging
+
+  beforeEach(function() {
+    serviceFactory = createServiceFactory()
+    configService = serviceFactory.getService('ConfigService')
+    apmServer = serviceFactory.getService('ApmServer')
+    performanceMonitoring = serviceFactory.getService('PerformanceMonitoring')
+    errorLogging = serviceFactory.getService('ErrorLogging')
+    configService.setConfig({
+      context: {
+        tags: {
+          foo: 'bar'
+        }
+      }
+    })
+  })
+
+  function getMappedKeys(keys) {
+    /**
+     * Some keys needs special mapping as they are used it only for
+     * internal data structures
+     */
+    const special = {
+      spans: 'y',
+      breakdown: 'me'
+    }
+    return keys.map(key => {
+      const mappedKey = special[key] || V3_MAPPING[key]
+      if (mappedKey != null) {
+        return mappedKey
+      }
+      return key
+    })
+  }
+
+  function isOptional(key) {
+    /**
+     * List of keys that are made optional in V3 SPEC
+     */
+    const OPTIONAL_KEYS = [
+      'transaction_id',
+      'parent_id',
+      'trace_id',
+      // Optional keys inside breakdown - breakdown.*.samples.<key>
+      'transaction-name',
+      'transaction-type',
+      'span-subtype'
+    ]
+    return OPTIONAL_KEYS.some(optKey => key.indexOf(optKey) >= 0)
+  }
+
+  function testMappedObject(original, compressed) {
+    const originalFlat = flatten(original)
+    const compressedFlat = flatten(compressed)
+
+    for (const key of Object.keys(originalFlat)) {
+      const keys = key.split(SEPARATOR)
+      const mappedKeys = getMappedKeys(keys)
+      const compressedKey = mappedKeys.join(SEPARATOR)
+      /**
+       * Some fields are made optional in the v3 spec so we check
+       * the condition only when the keys are not optional and
+       * values exist in the new compressed model
+       */
+      if (!isOptional(key) && compressedFlat[compressedKey] != null) {
```

Comment:
I think this should be `!isOptional(key) || compressedFlat[compressedKey] != null`

Reply:
damm. my bad 😃